main.go

```diff
@@ -50,6 +53,27 @@ func init() { //nolint:gochecknoinits
 	//+kubebuilder:scaffold:scheme
 }
 
+func setupReconciler(ctx context.Context, cli client.Client, mgr ctrl.Manager, clusterInfo *cluster.ClusterInfo) error {
```

Comment:
was this function just factored out from `main()`?

Reply:
Yes, to avoid linter error of [cyclomatic](https://pkg.go.dev/github.com/fzipp/gocyclo) complexity